### ISSUE TITLE
Checkbox component is missing adapter APIs, leading to `aria-checked` not being updated

### ISSUE THREAD

**Issue by kfranqueiro** [material-components/material-components-web#2579]
### What MDC Web Version are you using?

0.34.1

### What browser(s) is this bug affecting?

Any

### What OS are you using?

Any

### What are the steps to reproduce the bug?

* Go to http://material-components-web.appspot.com/checkbox.html
* Fiddle with indeterminate checkboxes
* Look at the `input` element

### What is the expected behavior?

`aria-checked` should be set to `mixed` on indeterminate JS checkboxes

### What is the actual behavior?

`aria-checked` is never set

### Any other information you believe would be useful?

Somehow we never implemented the `setNativeControlAttr` and `removeNativeControlAttr` adapter APIs in the checkbox component.

**Comment by kfranqueiro** [comment]
Traced to #2389, which added this functionality but didn't hook it up through the component, so it wasn't actually working.
